xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/Window.h

```diff
@@ -227,6 +481,135 @@ namespace gui
+    /// @param[in] value      string or unicode - value of property.
+    ///
+    /// @note Key is NOT case sensitive.\n
+    /// You can use the above as keywords for arguments and skip certain
+    /// optional arguments.\n
+    /// Once you use a keyword, all following arguments require the keyword.
+    ///
+    ///
+    void SetContainerProperty(const std::string& key, const std::string& value)
+    {
+      m_interface->kodi_gui->window->set_container_property(m_interface->kodiBase, m_controlHandle, key.c_str(), value.c_str());
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_CWindow
+    /// @brief Sets the content type of the container.
+    ///
+    /// @param[in] value      string or unicode - content value.
+    ///
+    /// __Available content types__
+    /// | Name        | Media                                    |
+    /// |:-----------:|:-----------------------------------------|
+    /// | actors      | Videos
+    /// | addons      | Addons, Music, Pictures, Programs, Videos
+    /// | albums      | Music, Videos
+    /// | artists     | Music, Videos
+    /// | countries   | Music, Videos
+    /// | directors   | Videos
+    /// | files       | Music, Videos
+    /// | games       | Games
+    /// | genres      | Music, Videos
+    /// | images      | Pictures
+    /// | mixed       | Music, Videos
+    /// | movies      | Videos
+    /// | Musicvideos | Music, Videos
+    /// | playlists   | Music, Videos
+    /// | seasons     | Videos
+    /// | sets        | Videos
+    /// | songs       | Music
+    /// | studios     | Music, Videos
+    /// | tags        | Music, Videos
+    /// | tvshows     | Videos
+    /// | videos      | Videos
+    /// | years       | Music, Videos
+    ///
+    ///
+    void SetContainerContent(const std::string& value)
+    {
+      m_interface->kodi_gui->window->set_container_content(m_interface->kodiBase, m_controlHandle, value.c_str());
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_CWindow
+    /// @brief Get the id of the currently visible container.
+    ///
+    /// @return               currently visible container id
+    ///
+    ///
+    int GetCurrentContainerId()
+    {
+      return m_interface->kodi_gui->window->get_current_container_id(m_interface->kodiBase, m_controlHandle);
+    }
+    //--------------------------------------------------------------------------
+    //@}
+
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_CWindow
+    /// @brief To inform Kodi that it need to render region new.
+    ///
+    ///
+    void MarkDirtyRegion()
```

Comment:
No Problem :)
With the documentation is currently added what I done some time ago. Currently only present but not direct adjusted for doxygen. Because of the size comes the optimization on separate requests.